Tests/ConfidenceTests/EventSenderEngineTest.swift

```diff
@@ -55,4 +55,30 @@ final class EventSenderEngineTest: XCTestCase {
         XCTAssertNil(uploader.calledRequest)
         cancellable.cancel()
     }
+
+    func testRemoveEventsFromStorageOnBadRequest() async throws {
```

Comment:
what are we testing here? if we remove the engine even the storage will have zero events (isEmpty will be true).

we are not sending events with engine here. therefore the storage won't have any events and the test will always pass. what we want to test is if we send events using engine, on bad request the storage cleans up the events.

Reply:
Ah, gocha! Thanks for pointing this out